src/main/java/org/isf/accounting/gui/PatientBillEdit.java

```diff
@@ -805,148 +923,120 @@ private JScrollPane getJScrollPaneTotal() {
+			Dimension size = new Dimension(PANEL_WIDTH, PAYMENT_HEIGHT);
+			jScrollPanePayment.setMaximumSize(size);
+			jScrollPanePayment.setMinimumSize(size);
+			jScrollPanePayment.setPreferredSize(size);
 		}
 		return jScrollPanePayment;
 	}
 
 	private JTable getJTablePayment() {
 		if (jTablePayment == null) {
-			jTablePayment = new JTable();
-			jTablePayment.setModel(new PaymentTableModel());
+			jTablePayment = new JTable(new PaymentTableModel());
 			jTablePayment.getColumnModel().getColumn(1).setMinWidth(PRICE_WIDTH);
 			jTablePayment.getColumnModel().getColumn(1).setMaxWidth(PRICE_WIDTH);
 		}
 		return jTablePayment;
 	}
 
+	private void setJTablePayment() {
+		jTablePayment.setModel(new PaymentTableModel());
+	}
+
 	private JScrollPane getJScrollPaneBalance() {
 		if (jScrollPaneBalance == null) {
 			jScrollPaneBalance = new JScrollPane();
 			jScrollPaneBalance.setViewportView(getJTableBalance());
 			jScrollPaneBalance.setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_NEVER);
-			jScrollPaneBalance.setMaximumSize(new Dimension(PANEL_WIDTH, BALANCE_HEIGHT));
-			jScrollPaneBalance.setMinimumSize(new Dimension(PANEL_WIDTH, BALANCE_HEIGHT));
-			jScrollPaneBalance.setPreferredSize(new Dimension(PANEL_WIDTH, BALANCE_HEIGHT));
+			jScrollPaneBalance.setMaximumSize(PatientBillEdit.BALANCE_TABLE_SIZE);
+			jScrollPaneBalance.setMinimumSize(PatientBillEdit.BALANCE_TABLE_SIZE);
+			jScrollPaneBalance.setPreferredSize(BALANCE_TABLE_SIZE);
 		}
 		return jScrollPaneBalance;
 	}
 
 	private JTable getJTableBalance() {
 		if (jTableBalance == null) {
-			jTableBalance = new JTable();
-			jTableBalance.setModel(new DefaultTableModel(new Object[][] {
-					{
-							"<html><b>"+MessageBundle.getMessage("angal.newbill.balance.txt").toUpperCase()+"</b></html>",
-							currencyCod,
-							balance}
-			},
-					new String[] {"","",""}) {
-				private static final long serialVersionUID = 1L;
-				Class<?>[] types = new Class<?>[] { JLabel.class, JLabel.class, Double.class, };
-
-				@Override
-				public Class<?> getColumnClass(int columnIndex) {
-					return types[columnIndex];
-				}
-
-				@Override
-				public boolean isCellEditable(int row, int column) {
-					return false;
-				}
-			});
-			jTableBalance.getColumnModel().getColumn(1).setMinWidth(CURRENCY_CODE_WIDTH);
-			jTableBalance.getColumnModel().getColumn(1).setMaxWidth(CURRENCY_CODE_WIDTH);
-			jTableBalance.getColumnModel().getColumn(2).setMinWidth(PRICE_WIDTH);
-			jTableBalance.getColumnModel().getColumn(2).setMaxWidth(PRICE_WIDTH);
-			jTableBalance.setMaximumSize(new Dimension(PANEL_WIDTH, BALANCE_HEIGHT));
-			jTableBalance.setMinimumSize(new Dimension(PANEL_WIDTH, BALANCE_HEIGHT));
-			jTableBalance.setPreferredSize(new Dimension(PANEL_WIDTH, BALANCE_HEIGHT));
+			jTableBalance = new JTable(new JTableBalanceModel());
+			sizeJTableBalance();
 		}
 		return jTableBalance;
 	}
 
+	private void setJTableBalance() {
```

Comment:
Why was this `private` method added as there are no uses

Reply:
it is used in `updateGUI()`

Reply:
Sorry, is should be `private JTableBalanceModel(Object[][] data, Object[] columnNames)` has no uses